indico/web/client/js/react/components/DateRangePicker.jsx

```diff
@@ -0,0 +1,224 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import DatePickerCalendar from 'indico/react/components/DatePickerCalendar';
+import {FinalField, validators as v} from 'indico/react/forms';
+import {Param, Translate} from 'indico/react/i18n';
+import {DateRange} from 'indico/utils/date';
+import {formatDate} from 'indico/utils/date_format';
+import {fromISOLocalDate} from 'indico/utils/date_parser';
+
+import 'indico/custom_elements/ind_date_picker';
+
+const INVALID = '__invalid__';
+
+export default function DateRangePicker({
+  value,
+  label,
+  format,
+  rangeStartLabel,
+  rangeEndLabel,
+  rangeStartMin,
+  rangeStartMax,
+  rangeEndMin,
+  rangeEndMax,
```

Comment:
We have another case where the callback is more convenient, and that's in the `WTFDateField` which can be linked to another field. While technically that's a date range, those older forms do not have the concept of a date range but it's one field per value (`start_date` and `end_date`).

Since we'd of course like to get rid of the old react-dates date picker (it prevents us from updating React to a recent version...), those fields will also need to be adapted to the new date picker, but without the callback that would be VERY painful or impossible. Here's the logic we use there:


```js
  const isOutsideRange = useCallback(
    value => {
      if (linkedField) {
        const linkedDate = moment(linkedFieldDateElem.value, 'DD/MM/YYYY');
        if (
          (linkedField.notBefore && value.isBefore(linkedDate, 'day')) ||
          (linkedField.notAfter && value.isAfter(linkedDate, 'day'))
        ) {
          return true;
        }
      }
      return (
        (earliestMoment && value.isBefore(earliestMoment, 'day')) ||
        (latestMoment && value.isAfter(latestMoment, 'day'))
      );
    },
    [earliestMoment, latestMoment, linkedField, linkedFieldDateElem]
  );
```